# <font color = black>Importing all the required modules</font>

In [ ]:
import pandas as pd
from matplotlib import pyplot as plt
import numpy as np
import matplotlib
matplotlib.rcParams["figure.figsize"] = (20,20)

# <font color = black>Importing the Property dataset</font>

In [5]:
df =pd.read_csv("property_data.csv")

In [6]:
df.head()

area  bathroom_num  bedroom_num    city  \
0  350.0           2.0            1  Mumbai   
1  652.0           2.0            1  Mumbai   
2  635.0           2.0            1  Mumbai   
3  540.0           2.0            1  Mumbai   
4  625.0           1.0            1  Mumbai   

                                                desc          dev_name  \
0  2 Bath,Unfurnished,East facing The project has...               NaN   
1  2 Bath,Semi-Furnished,East facing A 1BHK apart...       Veena Group   
2  2 Bath,Semi-Furnished,4 floor,West facing A be...     Agarwal Group   
3  2 Bath,Semi-Furnished,East facing Essential Se...               NaN   
4  1 Bath,Furnished,2 floor,North facing 24 hours...  Millennium Group   

   floor_count  floor_num      furnishing        id  ...  longitude  \
0          NaN        NaN     Unfurnished  45349857  ...  72.825882   
1          NaN        NaN  Semi-Furnished  45960973  ...  72.833592   
2          7.0        4.0  Semi-Furnished  46688849  ...  72.801612   
3          NaN        NaN  Semi-Furnished  44696119  ...  72.836006   
4          7.0        2.0       Furnished  46742851  ...  72.850167   

    post_date       poster_name  price            project  \
0  2020-01-11     Saurabh Patil   9000                NaN   
1  2020-01-11    Gorakh  Chavan   8060      Veena Dynasty   
2  2019-12-13      Lokesh Joshi   8000  Agarwal Lifestyle   
3  2020-01-13    Sajith  Thomas   8000                NaN   
4  2019-12-17  Charmaine  Gomez   9000     Mittal Enclave   

                                               title  trans       type  \
0             1 BHK Apartment for Rent in Malad West   Rent  Apartment   
1   1 BHK Apartment for Rent in Veena Dynasty, Vasai   Rent  Apartment   
2  1 BHK Apartment for Rent in Agarwal Lifestyle,...   Rent  Apartment   
3             1 BHK Apartment for Rent in Vasai East   Rent  Apartment   
4  1 BHK Apartment for Rent in Mittal Enclave, Na...   Rent  Apartment   

                                                 url user_type  
0  /propertyDetails/1-BHK-350-Sq-ft-Multistorey-A...     Agent  
1  /propertyDetails/1-BHK-652-Sq-ft-Multistorey-A...     Agent  
2  /propertyDetails/1-BHK-635-Sq-ft-Multistorey-A...     Agent  
3  /propertyDetails/1-BHK-540-Sq-ft-Multistorey-A...     Agent  
4  /propertyDetails/1-BHK-625-Sq-ft-Multistorey-A...     Owner  

[5 rows x 23 columns]

## <font color = black>Deleting the unnecessary columns and creating a new data set with only the important columns.</font>

In [7]:
df_final = df[["area","bathroom_num","bedroom_num","locality","price"]]
df_final.head()

area  bathroom_num  bedroom_num      locality  price
0  350.0           2.0            1    Malad West   9000
1  652.0           2.0            1         Vasai   8060
2  635.0           2.0            1         Virar   8000
3  540.0           2.0            1    Vasai East   8000
4  625.0           1.0            1  Naigaon East   9000

In [8]:
df_final.shape

(34348, 5)

# <font color = black>Checking for the null values if any.</font>

In [9]:
df_final.isnull().sum()

area            776
bathroom_num     14
bedroom_num       0
locality        208
price             0
dtype: int64

# <font color = black>Dropping all the Null values. Since our dataset has got 32000 rows, deleting the 208  null values wont have any impact.</font>

In [10]:
df_final1 = df_final.dropna()
df_final1.isna().sum()

area            0
bathroom_num    0
bedroom_num     0
locality        0
price           0
dtype: int64

In [11]:
df_final1.shape

(33355, 5)

In [12]:
df_final1["bedroom_num"].unique()

array([1, 3, 4, 5, 2], dtype=int64)

In [13]:
df_final1["bathroom_num"].unique()

array([2., 1., 3., 4., 5., 6., 7., 8.])

In [14]:
df_final1["locality"].unique()

array(['Malad West', 'Vasai', 'Virar', 'Vasai East', 'Naigaon East',
       'Virar West', 'Royal Palms Estate', 'Nalasopara', 'Silver Park',
       'Vasai West', 'Andheri East', 'Mulund East', 'Nalasopara East',
       'Balkum Pada - Majiwada', 'Kalyan', 'Bhayandar East',
       'Mira Bhayandar', 'MHADA Colony', 'Chembur Camp', 'Govind Nagar',
       'Bhayandar West', 'Beverly Park', 'Worli', 'Nalasopara West',
       'Lok Manya Nagar', 'Borivali West', 'Malvani', 'Viva College Road',
       'Mira Road', 'New Link Road', 'Sakinaka', 'Dongri', 'MMRDA Area',
       'Yashavant Nagar', 'Kandivali East', 'Ghatkopar West',
       'Ghodbunder', 'Nahur West', 'CAMA Industrial Estate',
       'Anushakti Colony', 'Thane', 'Yari Road', 'Virar East',
       'Goregaon East', 'Prabhadevi', 'Andheri West', 'Pandurang Wadi',
       'Bandra West', 'Khar West', 'Bolinj', 'Virat Nagar',
       'Thane-Kalyan-Dombivli', 'Poonam Gardens', 'Ramdev Park',
       'Vinay Nagar', 'Unique Garden', 'Tirupati Nagar

In [81]:
df_final_copy = df_final1.copy()

# <font color = black>Deleting terms like "West" and "East" from the address of the property. The below function is written for the same.</font>

In [64]:
def local(x):
    name = ""
    if "west" in x.lower() or "east" in x.lower():
        a = x.split()
        b = a.pop()
        for s in a:
            name = name+s
        return name
    else:
        return x

# <font color = black>Cheking if the function works properly before applying it on the dataset.</font>

In [65]:
print(local("Borivali East"))

Borivali


# <font color = black>Applying the function on our dataset</font>

In [77]:
df_final_copy["locality"] = df["locality"].apply(lambda x:local(str(x)))
df_final_copy["locality"].unique()

array(['Malad', 'Vasai', 'Virar', 'Naigaon', 'Royal Palms Estate',
       'Nalasopara', 'Silver Park', 'Andheri', 'Mulund',
       'Balkum Pada - Majiwada', 'Kalyan', 'Bhayandar', 'Mira Bhayandar',
       'MHADA Colony', 'Chembur Camp', 'Govind Nagar', 'Beverly Park',
       'Worli', 'Lok Manya Nagar', 'Borivali', 'Malvani',
       'Viva College Road', 'Mira Road', 'New Link Road', 'Sakinaka',
       'Dongri', 'MMRDA Area', 'Yashavant Nagar', 'Kandivali',
       'Ghatkopar', 'Ghodbunder', 'Nahur', 'CAMA Industrial Estate',
       'Anushakti Colony', 'Thane', 'Yari Road', 'Goregaon', 'Prabhadevi',
       'Pandurang Wadi', 'Bandra', 'Khar', 'Bolinj', 'Virat Nagar',
       'Thane-Kalyan-Dombivli', 'Poonam Gardens', 'Ramdev Park',
       'Vinay Nagar', 'Unique Garden', 'Tirupati Nagar',
       'Charkop Sector 8', 'Bandra Kurla Complex', 'Aarey Milk Colony',
       'Bhandup Industrial Estate', 'Oberoi Splendor', 'Charkop Village',
       'Devi Pada', 'Datta Pada', 'Jankalyan Nagar', 'Magath

In [80]:
df_final_copy["area"].unique()

array([350., 652., 635., ..., 150., 663., 642.])

In [85]:
df_final_copy.to_csv("Properties_cleaned.csv")